boot/bootdata/hivesft.inf

```diff
@@ -457,9 +457,7 @@ HKLM,"SOFTWARE\Microsoft\Windows\CurrentVersion\Shell Extensions",,0x00000012
 HKLM,"SOFTWARE\Microsoft\Windows\CurrentVersion\ShellServiceObjectDelayLoad",,0x00000012
 
 ; FIXME - Setup doesn't handle extra paths
-HKLM,"SOFTWARE\Microsoft\Windows\CurrentVersion\RunOnce","",0x00000000,"cmd /c md ""%programfiles%\Internet Explorer\"" && move %windir%\iexplore.exe ""%programfiles%\Internet Explorer\"" && ""%programfiles%\Internet Explorer\iexplore.exe"" /RegServer"
-; Create .NET Framework InstallRoot key, reg_sz & full path
-HKLM,"SOFTWARE\Microsoft\Windows\CurrentVersion\RunOnce","InstallRoot.NET",0x00000000,"cmd /c reg add HKLM\SOFTWARE\Microsoft\.NETFramework /v InstallRoot /t REG_SZ /d %SystemRoot%\Microsoft.NET\Framework\"
+HKLM,"SOFTWARE\Microsoft\Windows\CurrentVersion\RunOnceEx\Finalize","1",,"||RUNDLL32.EXE SETUPAPI.DLL,InstallHinfSection Finalize 128 %systemroot%\inf\syssetup.inf"
```

Comment:
Hmh... I'll check whether the standard bootcd 2nd-stage setup already runs that thing, basically.
The only time where this runonce would be needed would be for the livecd.... if we were able, afterwards, to "delete" the iexplore.exe file from windir in the livecd :joy:

Reply:
Ideally we _could_ use the syntax: `setupapi.dll|InstallHinfSection|Finalize 128 %systemroot%\inf\syssetup.inf` (with the last part maybe that requires quotes...); however our RunOnceEx doesn't seem to support that yet.

Reply:
> I'll check whether the standard bootcd 2nd-stage setup already runs that thing, basically.

What do you mean?  The cmd /c ... command this replaces is the thing the bootcd uses to move and register iexplore.exe on first boot.

Reply:
Ah, actually I've found something, in windows' syssetup.inf:
In its `[Infs.Always]` section it has a line: `ie.inf,DefaultInstall`.
This will run the `ie.inf` file, and that latter one is the component installation file for Internet Explorer: exactly what we need. So I could suggest you move what you've added in syssetup.inf into a ie.inf file.
If you don't feel like doing it in this PR, no problem, this was just for your information.

Reply:
Infs.Always is executed during setup phase 2? Does ROS have a valid CSIDL_PROGRAMFILES directory at that point?

I can add a FIXME comment to the current PR...

Reply:
> Ideally we _could_ use the syntax

Absolutely. `RunOnceExEntry::Exec` current just skips the two `||`s and treats everything as a CreateProcess which is unfortunate. We have to inspect all uses of RunOnceEx in ROS before fixing this bug.

Reply:
`Infs.Always` is executed by syssetup.dll's `InstallSysSetupInfComponents()`, in turn called by `CommonInstall()`.
That other function is executed by either `InstallReactOS()` after it has created Program Files (via `InitializeProgramFilesDir()`) and the generic user profiles directory. It seems to be also executed in livecd mode from `InstallLiveCD()`.